pkg/storage/crd/delete.go

```diff
@@ -0,0 +1,51 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crd
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/golang/glog"
+	"k8s.io/apiserver/pkg/storage"
+	restclient "k8s.io/client-go/rest"
+)
+
+func delete(cl restclient.Interface, key, resource, ns, name string, hasNamespace bool, expectedCode int) error {
+	req := cl.Delete().Resource(resource).Name(name)
+	if hasNamespace {
+		req = req.Namespace(ns)
+	}
+	res := req.Do()
+	if res.Error() != nil {
+		glog.Errorf("executing DELETE for %s/%s (%s)", ns, name, res.Error())
```

Comment:
Shouldn't this be a `return fmt.Errorf(....)`? Why log anything when this function can return the error?
I guess you've been following the existing style, but this is can be improved, right?

Reply:
Think about it - there is no way caller gets any information about this `err`. This is not right.

p.s. same in other files, will not comment there.